rules/preprocessing.smk

```diff
@@ -1,6 +1,45 @@
+NUM_BATCHES        = config["num_batches"]
```

Comment:
I don't know about this one, because all Snakemake files using this style of spacing to pass values from config file.

Reply:
> because all Snakemake files using this style of spacing

Yes, and they also should be refactored (one day we'll do that!)
https://peps.python.org/pep-0008/#whitespace-in-expressions-and-statements